src/OpenTelemetry/Resources/OtelEnvResourceDetector.cs

```diff
@@ -31,7 +31,7 @@ public Resource Detect()
         return resource;
     }
 
-    private static IEnumerable<KeyValuePair<string, object>> ParseResourceAttributes(string resourceAttributes)
```

Comment:
why this change?

The returned list is used in the Resource ctor
https://github.com/open-telemetry/opentelemetry-dotnet/blob/0252f6ca0dfe66c0a2b6251218423ae0760d92ff/src/OpenTelemetry/Resources/OtelEnvResourceDetector.cs#L27-L28

Resource ctor expects an IEnumerable
https://github.com/open-telemetry/opentelemetry-dotnet/blob/0252f6ca0dfe66c0a2b6251218423ae0760d92ff/src/OpenTelemetry/Resources/Resource.cs#L22

Reply:
I agree, but analyzer is pretty strict here. All private methods should return the most concrete possible type.
I think that it is better to fix it, than commenting out this warning.

This construction also does not solve the issue.
```
            resource = new Resource(ParseResourceAttributes(envResourceAttributeValue));
```

https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1859